packages/expo-router/src/modal/Modal.tsx

```diff
@@ -110,7 +118,7 @@ export function Modal(props: ModalProps) {
     if (!areDetentsValid(detents)) {
       throw new Error(`Invalid detents provided to Modal: ${JSON.stringify(detents)}`);
     }
-  }, [detents]);
+  }, [detents, detentIndex]);
```

Comment:
This is not needed here, as it's not checked in the `useEffect`. However I think it will be good to add a check that will only work when `__DEV__` is true. Otherwise people will see screens warning which may be confusing 

https://github.com/software-mansion/react-native-screens/blob/main/src/components/helpers/sheet.tsx#L95